File: library/src/main/java/com/anthonycr/bonsai/ReactiveEventException.java
package com.anthonycr.bonsai;

/**
 * An exception thrown by bonsai when problems caused by misuse of the library arise.
 */
class ReactiveEventException extends Exception {
    public ReactiveEventException(String message) {
        super(message);
    }

    public ReactiveEventException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: library/src/main/java/com/anthonycr/bonsai/refactor/ReactiveEventException.kt
package com.anthonycr.bonsai.refactor

/**
 * Created by anthonycr on 9/9/17.
 */

internal class ReactiveEventException : Exception {

    constructor(message: String) : super(message)

    constructor(message: String, cause: Throwable) : super(message, cause)

}